lib/bibdata_rs/src/theses/holdings.rs

```diff
@@ -0,0 +1,285 @@
+// This module is responsible for describing the holdings of a thesis
+
+use crate::theses::embargo;
+use serde::{ser::SerializeStruct, Serialize};
+use std::collections::HashMap;
+
+pub fn call_number(non_ark_ids: Option<&Vec<String>>) -> String {
+    let ids = match non_ark_ids {
+        Some(value) => value,
+        None => &Vec::default(),
+    };
+    match ids.first() {
+        Some(id) => format!("AC102 {}", id),
+        None => "AC102".to_string(),
+    }
+}
+
+pub fn online_holding_string(non_ark_ids: Option<&Vec<String>>) -> Option<String> {
+    serde_json::to_string(&ThesisHoldingHash {
+        thesis: OnlineHolding {
+            call_number: call_number(non_ark_ids),
+        },
+    })
+    .ok()
+}
+
+pub fn physical_holding_string(non_ark_ids: Option<&Vec<String>>) -> Option<String> {
+    serde_json::to_string(&ThesisHoldingHash {
+        thesis: PhysicalHolding {
+            call_number: call_number(non_ark_ids),
+        },
+    })
+    .ok()
+}
+
+fn physical_class_year(class_years: &[String]) -> bool {
+    // For theses, there is no physical copy since 2013
+    // anything 2012 and prior have a physical copy
+    // @see https://github.com/pulibrary/orangetheses/issues/76
+    match class_years.first() {
+        Some(year) => year < &"2013".to_string(),
+        None => false,
+    }
+}
+
+pub fn ark_hash(
```

Comment:
This is mapped into one of the electronic display fields (we think) - maybe a different name?